client/mapView/MapView.h

```diff
@@ -49,7 +49,14 @@ class MapView : public BasicMapView
 {
 	std::shared_ptr<MapViewActions> actions;
 
+	std::map<uint32_t, Point> swipeHistory;

```

Comment:
not quite optimal to use std::map. If I understand logic here properly, you insertion to the end, iterating over element and access to last element. If so, the best performance would give std::list

```suggestion
	std::list<std::pair<uint32_t, Point>> swipeHistory;
```

Reply:
List is *almost never* optimal. Exceptions do exist but they are extremely rare.
If you want something with better performance then you need vector. Hard to beat its guaranteed memory locality.

But since this is not performance-critical code using most convenient containers is better idea than most optimal one.

Reply:
Changed to vector